vitrina/projects/forms.py

```diff
@@ -45,3 +48,70 @@ def __init__(self, *args, **kwargs):
             Field("image"),
             Submit("submit", button, css_class="button is-primary"),
         )
+
+class ClientCreateForm(ModelForm):
+    name = CharField(label=_("Pavadinimas"), help_text=_("Kliento pavadinimas"))
+
+    class Meta:
+        model = UseCaseClient
+        fields = ["name"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        project_instance = self.instance if self.instance and self.instance.pk else None
+        button = _("Redaguoti") if project_instance else _("Sukurti")
+        self.helper = FormHelper()
+        self.helper.attrs["novalidate"] = ""
+        self.helper.form_id = "client-form"
+        self.helper.layout = Layout(
+            Field("name", placeholder=_("Pavadinimas")),
+            Submit("submit", button, css_class="button is-primary"),
+        )
+
+
+class ClientScopeCreateForm(ModelForm):
+    scope = ModelChoiceField(
+        queryset=AgreementScope.objects.none(),
+        label=_("Leidimas"),
+        help_text=_("Pasirinkite leidimą"),
+    )
+
+    class Meta:
+        model = UseCaseClientScope
+        fields = ["scope"]
+
+    def __init__(self, *args, available_scopes: UseCaseClientScope = None, use_case_client: UseCaseClient = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        if available_scopes and use_case_client:
+            existing_scopes = set(
+                UseCaseClientScope.objects
+                .filter(use_case_client=use_case_client)
+                .values_list("resource", flat=True)
+            )
+
+            filtered_scopes = [
+                scope for scope in available_scopes
+                if scope.resource not in existing_scopes
+            ]
+            self.fields["scope"] = ModelChoiceField(
+                queryset=AgreementScope.objects.filter(pk__in=[s.pk for s in filtered_scopes]),
+                label=_("Leidimas"),
+                help_text=_("Pasirinkite leidimą")
+            )
```

Comment:
```suggestion
    self.fields["scope"].queryset = available_scopes
```
You don't need to filter scopes in python and rebuild whole scope `ModelChoiceField`. Changing queryset is enough.

And if you get this queryset in view, you can also check if no available scopes exists. If not - don't load this form at all. If you do this, you can also make `scope` field required and remove `clean_scope` method.